### DISCUSSION TITLE
malformed response while connecting to api via webhook payload URL in github

### DISCUSSION THREAD

**Discussion by Satyavanich** [community/community#55723] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

may i know the reason for the below error and how to rectify this error
error is occurred while i am sending webhook payload to post api(api internal to IBM) via IBM github webhooks

![Screenshot 2023-05-16 150025](https://github.com/community/community/assets/127304437/6f8e78f8-1ed5-4a2e-a79c-c35eb15a21af)

**Comment by airtower-luna** [comment]
That error indicates a problem with that internal API server. The webhook sends an HTTP request, and the server needs to send a valid HTTP response. According to the error message it didn't do that.

An [example in the documentation](https://docs.github.com/en/webhooks-and-events/webhooks/testing-webhooks) simply uses a 200 response with an empty body, I'd probably try 204 ("No Content") for that, but the server needs to send an HTTP response.

**Comment by Satyavanich** [comment]
Thank you for your response ,
but when i am trying in talend api tester with the same post api and github request i am able to get the response from server
![Screenshot 2023-05-18 153625](https://github.com/community/community/assets/127304437/f56b4bda-290d-41ff-bdab-0a43baff4728)
May i know why this error is getting in github

**Comment by airtower-luna** [comment]
I can't answer that. If the server treats the requests differently, you'll have to check on the server why. For an internal service, maybe there's a firewall in the way?
